src/Provider/AbstractProvider.php

```diff
@@ -702,18 +702,26 @@ protected function parseResponse(ResponseInterface $response)
      * Custom mapping of expiration, etc should be done here. Always call the
      * parent method when overloading this method.
      *
-     * @param  mixed $result
-     * @return array
+     * @param  array|string $response
+     * @return array|string
+     * @throws UnexpectedValueException
      */
-    protected function prepareAccessTokenResponse(array $result)
+    protected function prepareAccessTokenResponse($response)
```

Comment:
> In the case of prepareAccessTokenResponse, we absolutely know that the response must be an array

I agree with this, as well as everything else both you and @ramsey are saying. We know that it has to be an array, and the typehint should enforce that.

> Type hinting provides the same protections that throwing an exception does.

"If for whatever reason the value passed to prepareAccessTokenResponse is not an array, I think an UnexpectedValueException is more meaningful than the default PHP error."